src/main/java/cam72cam/mod/world/World.java

```diff
@@ -115,10 +115,13 @@ private void checkLoadedEntities() {
                     this.onEntityAdded(entity);
                 }
             }
-            for (Entity entity : new ArrayList<>(this.entityByID.values())) {
-                if (!this.internal.loadedEntityList.contains(entity.internal)) {
-                    ModCore.warn("Dropping entity that was not removed correctly %s - %s", entity.getUUID(), entity);
-                    this.onEntityRemoved(entity.internal);
+	    Iterator<Map.Entry<Integer,cam72cam.mod.entity.Entity>> entityIterator = this.entityByID.entrySet().iterator();
```

Comment:
The copy here was important due to concurrent modification exceptions

Reply:
You cant modify a list while iterating it; the Iterator is in fact a solution for this. Interator.remove() is safe agains ConcurrentModificationException.

Reply:
Daniel is correct. The iterator.remove() is CME-safe. Copies will incur an additional memory cost as well as a slight performance penalty. It's better to delete the entity from the other maps before calling iterator.remove() to remove it from the entityByID map itself.

Reply:
See updated comment.

Reply:
We'll probably need to use a ConcurrentHashMap. There's more info here from Oracle on the ConcurrentHashMap spec. It's part of java.util.concurrent and does not require any external dependency. I will profile ConcurrentHashMap vs HashMap as well as pit cam72cam's solution against mine to identify the most performant solution.